### PR TITLE
feat: Add window cover, add covers to readme

### PR DESCRIPTION
**Pull request by InTheDaylight14**
Tested window functionality based on my fork of teslajsonpy in the dev container. I was able to get it working. Addresses https://github.com/alandtse/tesla/issues/234

One note: I had to comment out lines with "CoverEntityFeature" in the dev container for the cover entities to load on setup (I assume that will work with future HA versions?). I left them intact for this PR.

### CODE REVIEW

**Comment by alandtse** [comment]
You need to bump teslajsonpy now. In the future link the dependency pr so I know.

**Comment by InTheDaylight14** [comment]
Looks like teslajsonpy 3.0.0 was being installed for the tests. Sorry, still new to this

File: tests/test_cover.py [hunk 1]
```diff
@@ -78,3 +78,26 @@ async def test_trunk(hass: HomeAssistant) -> None:
...
+            {ATTR_ENTITY_ID: "cover.my_model_s_window"},
+            blocking=True,
+        )
+        mock_open_cover.assert_awaited_once()
+
+    with patch("teslajsonpy.car.TeslaCar.close_windows") as mock_close_cover:
+        assert await hass.services.async_call(
+            COVER_DOMAIN,
+            SERVICE_CLOSE_COVER,
+            {ATTR_ENTITY_ID: "cover.my_model_s_windows"},
```
Comment (alandtse) [inline tests/test_cover.py:100]: Do you need to remove the s here too?

**Comment by InTheDaylight14** [comment]
I'm not sure why the test is failing. When running them on my local machine I'm getting a different error. All the tests fail with: 
``` pytest_socket.SocketBlockedError: A test tried to use socket.socket.```

Any guidance on what I can do to figure it out?

**Comment by alandtse** [comment]
> When running them on my local machine I'm getting a different error. All the tests fail with:
> ` pytest_socket.SocketBlockedError: A test tried to use socket.socket.`

I haven't looked at the tests, but in theory tests should not actually touch the network but instead should have the API response mocked so no API is actually touched.

**Comment by InTheDaylight14** [comment]
Could this be an issue when installing the dependencies in the tests? 
```
Warning: poetry.lock is not consistent with pyproject.toml. You may be getting improper dependencies. Run `poetry lock [--no-update]` to fix it.
```

I can commit the updated lock file.

**Comment by alandtse** [comment]
Yes, you need to run `poetry update` and commit the lock file whenever you change/update dependencies. However, that won't solve the issue if the mocks are hitting apis.

**Comment by alandtse** [comment]
It's failing because self.is_closed is `true` so it doesn't try to close it again.
https://github.com/alandtse/tesla/pull/318/files#diff-3ebb2677d043645911800a521a02da6fa68181b50949d10428055114372a9b5bR176

You'll notice that other covers don't have this requirement at all.

The issue is you're mocking the underlying teslajsonpy call, which is responsible with updating the car state. Because the call is not happening, HA's state isn't changing.

The way for the test to work is to set the car window to open and then run the service_call. You may want to look at HA cover tests to see the right way to set an entity's state. It's been a while since I looked at it.

**Comment by InTheDaylight14** [comment]
Thanks for the info, I think I'll be able to do it now. I'm going to start from the top if that's okay. This got messy

**Comment by alandtse** [comment]
For reference, the reason for the last change was explained here. https://github.com/alandtse/tesla/pull/338

**Comment by alandtse** [comment]
Is the API smart enough that trying to open or close a window when it's already open or closed isn't a problem for the car? Even though we have a warranty disclaimer, let's not purposely release something that can cause physical damage.

**Comment by InTheDaylight14** [comment]
> Is the API smart enough that trying to open or close a window when it's already open or closed isn't a problem for the car? Even though we have a warranty disclaimer, let's not purposely release something that can cause physical damage.

Are any other changes needed?

**Pull request merged**